core/src/graph/page.rs

```diff
@@ -57,7 +57,7 @@ impl<E: EncryptionBehavior> TryFrom<(&PageData, &Vec<KeyPair<E>>)> for GraphPage
 		// First try the key that was indicated in the page
 		if let Some(indicated_key) = keys.iter().find(|k| k.key_id == key_id) {
 			if let Ok(data) =
-				E::decrypt(&encrypted_compressed_private_graph, &indicated_key.private_key)
+				SealBox::decrypt(&encrypted_compressed_private_graph, &indicated_key.key_pair)
```

Comment:
replaced with concrete implementation but I think we should decide about which way we want to go based on previous comment.